client/sites/marketing/tools/index.tsx

```diff
@@ -20,17 +24,46 @@ import './style.scss';
 export default function MarketingTools() {
 	const translate = useTranslate();
 	const dispatch = useDispatch();
-	const recordTracksEvent = ( event: string ) => dispatch( recordTracksEventAction( event ) );
+	const [ searchTerm, setSearchTerm ] = useState( '' );
 	const selectedSiteSlug: T.SiteSlug | null = useSelector( getSelectedSiteSlug );
 	const siteId = useSelector( getSelectedSiteId ) || 0;
 
+	const recordTracksEvent = useCallback(
+		( event: string, properties?: Record< string, string > ) =>
+			dispatch( recordTracksEventAction( event, properties ) ),
+		[ dispatch ]
+	);
+
 	const marketingFeatures = getMarketingFeaturesData(
 		selectedSiteSlug,
 		recordTracksEvent,
 		translate,
 		localizeUrl
 	);
 
+	const marketingFeaturesFiltered = useMemo( () => {
+		return marketingFeatures.filter(
```

Comment:
Since dependencies might change ( `marketingFeatures` ), I would add a condition for `searchTerm !== ''` before filtering.

Reply:
I considered that scenario, empty string will return the whole array and we would avoid adding an extra condition there.

Reply:
Hmm, I thought about it again. We need it that way in case the search term is cleared.